### PR TITLE
Tests/gaze detection test

### PR DESCRIPTION
**Pull request by kuredoro**

### CODE REVIEW

**Review by alkaitagi — APPROVED** [review]

File: vpt/processors/gaze_engagement_estimator/__init__.py [hunk 1]
```diff
@@ -10,9 +10,16 @@ class GazeEngagementEstimator(ProcessorBase[np.ndarray]):
...
 
-    def __init__(self, rotation_vector_source: ProcessorBase[np.ndarray]):
+    def __init__(self, rotation_vector_source: ProcessorBase[np.ndarray], threshold=0.5):
+        self.threshold = threshold
         rotation_vector_source.get_data_stream().subscribe(lambda v:
-                                                             self.subj.on_next(np.linalg.norm(v)))
+                                                           self.subj.on_next(np.linalg.norm(v) < self.threshold))
+
+    def start(self):
+        pass
```
Comment (IamMaxim) [inline vpt/processors/gaze_engagement_estimator/__init__.py:20]: Ah, yes... Processors are indeed sources...

Comment (aabounegm) [reply vpt/processors/gaze_engagement_estimator/__init__.py:20]: What I don't understand is why isn't the default implementation for `start` in Processors just `self.subj.start()`

File: vpt/processors/gaze_engagement_estimator/__init__.py [hunk 2]
```diff
@@ -10,9 +10,16 @@ class GazeEngagementEstimator(ProcessorBase[np.ndarray]):
     '''Transforms the gaze estimation to engagement estimation.'''
     subj = Subject()
 
-    def __init__(self, rotation_vector_source: ProcessorBase[np.ndarray]):
+    def __init__(self, rotation_vector_source: ProcessorBase[np.ndarray], threshold=0.5):
```
Comment (aabounegm) [inline vpt/processors/gaze_engagement_estimator/__init__.py]: Is `rotation_vector_source` really a `ProcessorBase` or can it just be a `SourceBase`?
I think there are very few places where it would make sense to require a parameter to be a `Processor`. You usually just depend on the fact that it provides you with data, not that it also gets fed some data

File: vpt/tests/test_gaze_detector.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
+import sys
+# Test supposed to run as python -m unittest path/to/file.py
+# from the root of repository
+# If you know better way than using this sys.path thingy, tell me
+sys.path.append('.')
```
Comment (aabounegm) [inline vpt/tests/test_gaze_detector.py:7]: Why is this needed?

File: vpt/tests/test_gaze_detector.py [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+import unittest
+import numpy as np
+import time
+from vpt.sources.video_file_source import VideoFileSource
+from vpt.processors.gaze_detector import GazeDetector
+from vpt.processors.gaze_engagement_estimator import GazeEngagementEstimator
+from vpt.sinks.video_display import VideoDisplay
+
+print(sys.path)
```
Comment (aabounegm) [inline vpt/tests/test_gaze_detector.py]: You probably forgot to remove this?

File: vpt/processors/gaze_engagement_estimator/__init__.py (hunk 1 above)
Comment (IamMaxim) [reply vpt/processors/gaze_engagement_estimator/__init__.py:20]: `subj` does not have any start methods, think of it as of UNIX pipe. Can you start a UNIX pipe?

Comment (aabounegm) [reply vpt/processors/gaze_engagement_estimator/__init__.py:20]: I probably meant the sources it depends on that are passed in the constructor, but then I realized that you cannot do that in the default implementation in the parent class.
But still, all `Processor` subclasses should override `start` to call `.start()` on all sources they depend on, not just `pass`

Comment (illright) [reply vpt/processors/gaze_engagement_estimator/__init__.py:20]: It sounds like a good solution, but let's not bother with this and just start all sources manually, please. This will bring no value to our project

**Review by illright — APPROVED** [review]

File: vpt/processors/gaze_engagement_estimator/__init__.py (hunk 1 above)
Comment (aabounegm) [reply vpt/processors/gaze_engagement_estimator/__init__.py:20]: Sure, but we're currently not even doing that. Most of the processors are currently black holes (they will not call `start` on their own sources) or even outright throw an exception.

Comment (illright) [reply vpt/processors/gaze_engagement_estimator/__init__.py:20]: Processors are not expected to call `start` on their sources, the way we currently have it. We just assume that a source will come bearing data

Comment (aabounegm) [reply vpt/processors/gaze_engagement_estimator/__init__.py:20]: yeah, well, that shouldn't be the case. It doesn't make much sense. From the perspective of sinks, processors are just sources, and calling `start` on sources should \[surprisingly enough\] _start_ them. How can they start? By asking their own sources to start so that they will actually have data

Comment (illright) [reply vpt/processors/gaze_engagement_estimator/__init__.py:20]: We don't need to call `start()` on processors. The reason why we have it is because of some bullshit with abstract classes that requires them

**Review by IamMaxim — APPROVED** [review]

File: vpt/tests/test_gaze_detector.py (hunk 3 above)
Comment (kuredoro) [reply vpt/tests/test_gaze_detector.py:7]: Well, uuuugh...  Yeah, I forgot to mention it. Oops... The problem is that it cannot find `vpt` module :(

File: vpt/tests/test_gaze_detector.py [hunk 5]
```diff
@@ -0,0 +1,73 @@
+import sys
+# If you know better way than using this sys.path thingy, tell me
```
Comment (aabounegm) [inline vpt/tests/test_gaze_detector.py:2]: This seems like the way it should be, although it can be a little bit cleaner by moving it to an `__init__.py` file as described in [this SO answer](https://stackoverflow.com/a/31489441). Also, don't you think the tests folder should be outside `vpt`? It is not supposed to be a distributable part of the package, and maybe that is part of the problem?

There also seems to be a [command line argument](https://docs.python.org/3/library/unittest.html#test-discovery) to set the starting directory for discovery

**Review by aabounegm — APPROVED** [review]
